lib/straight-server/websocket_insight_client.rb

```diff
@@ -0,0 +1,56 @@
+module StraightServer
+  class WebsocketInsightClient
+
+    @@address_check_list = []
+    
+    class << self
+      
+      def start(url)
+        socket = SocketIO::Client::Simple.connect url
+
+        socket.on :connect do
+          StraightServer.logger.info "Connected to Insight websocket with url: #{url}"
+          socket.emit :subscribe, 'inv'
+          @wclient = StraightServer::WebsocketInsightClient.new
+        end
+
+        socket.on :tx do |data|
+          @wclient.check_transaction(data) if data["vout"]
+        end
+
+        socket.on :error do |err|
+          StraightServer.logger.warn err
+        end
+      end
+
+      def address_for_check_list
+        @@address_check_list
+      end
+
+      def add_address(address)
+        @@address_check_list.push(address).uniq!
+        StraightServer.logger.info "[WS] Added address for tracking: #{@@address_check_list}"
+      end
+
+      def remove_address(address)
+        @@address_check_list.delete(address)
+      end
+
+      def clear_address_list
+        @@address_check_list = []
+      end
+    end
+
+    def check_transaction(data)
+      data["vout"].each do |o|
+        if index = @@address_check_list.find_index(o.keys.first)
+          address = @@address_check_list.delete_at(index)
+          StraightServer.logger.info "[WS] Found transaction for address: #{address}"
+          StraightServer::Order.set_status_for(address, data)
+        end
+      end
+    end
+    
+  end
+
+end
```

Comment:
Not exactly. Callback you pass as argument (in that case, add new address for monitoring). And then when some event happens (for example transaction detected) you call it. Which mean – who asked, that receives.
In pu/sub pattern, when event happens it just broadcast about it. And those who listen for that event, fire their methods. So no need to pass any additional data to event emiter and you can add as much listeners as you want in any places of application. 
Very good implementation of it is Whisper gem.
Here nice article (but related mostly to RoR): http://www.toptal.com/ruby-on-rails/the-publish-subscribe-pattern-on-rails